hasher-matcher-actioner/hmalib/models.py

```diff
@@ -59,51 +67,217 @@ class PipelinePDQHashRecord(PDQRecordBase):
+        records = [
+            PipelinePDQHashRecord(
+                item["PK"][len(DynamoDBItem.CONTENT_KEY_PREFIX) :],
+                item["ContentHash"],
+                datetime.datetime.fromisoformat(item["UpdatedAt"]),
+                item["Quality"],
+            )
+            for item in items
+        ]
+        return None if not records else records[0]
+
 
 @dataclass
 class PDQMatchRecord(PDQRecordBase):
     """
     Successful execution at the matcher produces this record.
     """
 
-    SIGNAL_KEY_PREFIX = "s#"
-    signal_id: int
+    signal_id: t.Union[str, int]
     signal_source: str
     signal_hash: str
 
     @staticmethod
     def get_dynamodb_signal_key(source: str, s_id: t.Union[str, int]) -> str:
         return f"{PDQMatchRecord.SIGNAL_KEY_PREFIX}{source}#{s_id}"
 
+    @staticmethod
+    def remove_signal_key_prefix(key: str, source: str) -> str:
+        return key[len(PDQMatchRecord.SIGNAL_KEY_PREFIX) + len(source) + 1 :]
+
     def to_dynamodb_item(self) -> dict:
         return {
-            "PK": self.get_dynamodb_content_key(self.content_key),
+            "PK": self.get_dynamodb_content_key(self.content_id),
             "SK": self.get_dynamodb_signal_key(self.signal_source, self.signal_id),
             "ContentHash": self.content_hash,
-            "Timestamp": self.timestamp.isoformat(),
+            "UpdatedAt": self.updated_at.isoformat(),
             "SignalHash": self.signal_hash,
             "SignalSource": self.signal_source,
             "GSI1-PK": self.get_dynamodb_signal_key(self.signal_source, self.signal_id),
-            "GSI1-SK": self.get_dynamodb_content_key(self.content_key),
+            "GSI1-SK": self.get_dynamodb_content_key(self.content_id),
             "HashType": self.SIGNAL_TYPE,
             "GSI2-PK": self.get_dynamodb_type_key(self.SIGNAL_TYPE),
         }
 
     def to_sqs_message(self) -> dict:
         # TODO add method for when matches are added to a sqs
         raise NotImplementedError
+
+    @classmethod
+    def get_from_content_id(
+        cls, table: Table, content_id: str
+    ) -> t.List["PDQMatchRecord"]:
+        items = MatchRecordQuery.from_content_key(
+            table,
+            cls.get_dynamodb_content_key(content_id),
+            cls.SIGNAL_KEY_PREFIX,
+            cls.SIGNAL_TYPE,
+        )
+        return cls._result_items_to_records(items)
+
+    @classmethod
+    def get_from_signal(
+        cls, table: Table, signal_id: t.Union[str, int], signal_source: str
+    ) -> t.List["PDQMatchRecord"]:
+        items = MatchRecordQuery.from_signal_key(
+            table,
+            cls.get_dynamodb_signal_key(signal_source, signal_id),
+            cls.SIGNAL_TYPE,
+        )
+        return cls._result_items_to_records(items)
+
+    @classmethod
+    def get_from_time_range(
+        cls, table: Table, start_time: str = None, end_time: str = None
+    ) -> t.List["PDQMatchRecord"]:
+        items = MatchRecordQuery.from_time_range(
```

Comment:
If we were to create a second MatchRecord subclass, do you believe that would have the same Projection String? We might have discussed this before, but would like to get your present understanding.

Reply:
Because if not, you might want to retain the ProjectionString as an attribute of this class rather than delegate to MatchRecordQuery.

Reply:
Good question, I wasn't sure. I think you're likely right but after going back and forth with where I wanted thas string to live I decided on in MatchRecordQuery for now, if we end up not being able to/wanting to reuse it I will definitely move it to the data class.